security/pkg/k8s/chiron/controller.go

```diff
@@ -205,8 +205,9 @@ func (wc *WebhookController) upsertSecret(secretName, dnsName, secretNamespace s
 		return nil
 	}
 
+	requestedLifetime := time.Duration(0)
```

Comment:
What does 0 mean?



It seems not quite right https://github.com/jetstack/cert-manager/blob/75d91bcb297d8a1506abd376ea378cf760c4f7f1/pkg/util/pki/csr.go#L380

Reply:
This is for Backward compatiblity : we will use  the `kubernetes.io/legacy-unknow' by default. In that case, no TTL can be specified.  So using a  '0' means we do not want to add related annotation to the CSR.  See : https://github.com/istio/istio/pull/34484/files#diff-6fa317e12eadb34d01780d845992c68fb30c869c9cc9127fd6458e728dacec1fR247

Reply:
make sense